File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/http/handler/ErrorHandlerFactory.java
package com.mcgrady.xskeleton.http.handler;

import android.content.Context;

import com.mcgrady.xskeleton.http.interf.ResponseErrorListener;

/**
 * Created by mcgrady on 2019/4/26.
 */
public class ErrorHandlerFactory {
    public final String TAG = this.getClass().getSimpleName();
    private Context context;
    private ResponseErrorListener mResponseErrorListener;

    public ErrorHandlerFactory(Context context, ResponseErrorListener responseErrorListener) {
        this.context = context;
        this.mResponseErrorListener = responseErrorListener;
    }

    public void handlerError(Throwable throwable) {
        mResponseErrorListener.handlerResponseError(context, throwable);
    }

}

File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/http/handler/ErrorHandlerFactory.kt
package com.mcgrady.xskeleton.http.handler

import android.content.Context
import com.mcgrady.xskeleton.http.interf.ResponseErrorListener

/**
 * Created by mcgrady on 2019/4/26.
 */
class ErrorHandlerFactory(private val context: Context, private val mResponseErrorListener: ResponseErrorListener) {
    val TAG = this.javaClass.simpleName
    fun handlerError(throwable: Throwable?) {
        mResponseErrorListener.handlerResponseError(context, throwable)
    }

    /**
     *
     * @param resultCode
     * @param errMsg
     * @return isSuccess
     */
    fun handlerResultCode(resultCode: Int, errMsg: String?): Boolean {
        return mResponseErrorListener.handlerErrorResultCode(context, resultCode, errMsg)
    }

}